impl/src/main/java/io/kroxylicious/testing/kafka/testcontainers/TestcontainersKafkaCluster.java

```diff
@@ -290,17 +300,17 @@ private synchronized String buildBrokerList(Function<Integer, KafkaClusterConfig
      * @param overrideVersion desired version, which may be null.
      * @return container image name
      */
-    private DockerImageName overrideContainerImageTagIfNecessary(@NonNull DockerImageName image, @Nullable String overrideVersion) {
+    private PerImagePullPolicy overrideContainerImageTagIfNecessary(@NonNull DockerImageName image, @Nullable String overrideVersion) {
         if (overrideVersion == null || overrideVersion.equalsIgnoreCase(image.getVersionPart())) {
```

Comment:
I wonder if we should allow the behaviour to be overridden by an env var in order to support use-cases such as disconnected developer environments.

Reply:
I'm inclined to agree, not sure here is the point to do so but I think the idea is good

Reply:
We support loading from ENV Vars a few levels higher see https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/0822c5f5b096b1feee58b2170bcedf100f466e8d/impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterFactory.java#L86-L91 

But those then get overridden by https://github.com/kroxylicious/kroxylicious-junit5-extension/blob/00b40bc9e436cdc119bd0948c2077a9781717314/impl/src/main/java/io/kroxylicious/testing/kafka/common/KafkaClusterConfig.java#L96-L104 which seems the wrong way round to me. 

I'm going to try and rationalise things a bit. 

@franvila can you remind me what we were trying to support when adding the env vars? As I want to make sure we still support that need.

Reply:
Passing Kafka/Zookeeper repo by env vars allows the user to pass whatever kafka image different from default kafka/zookeeper-native from Ozan Gunalp repo or using an older version from of those default images. 

If so, those images are passed to `TestcontainersKafkaCluster`.

TBH I don't recall why this method was added, but I found the [commit](https://github.com/franvila/kroxylicious-junit5-extension/commit/0822c5f5b096b1feee58b2170bcedf100f466e8d) done by @k-wall

Reply:
The use-case it supports is the testing with kafka-native builds from alternative repos (other than `quay.io/ogunalp/kafka-native`).   I use this mechanism when I am making changes in kafka-native that I want to try out with the Junit Ext before they are merged to kafka-native.

Reply:
It would probably make more sense if those image override env vars were a concern of TestcontainersKafkaCluster rather than the KafkaClusterFactory.    They should just override the images that would already be used.

Reply:
FFS I did that but apparently its not pushed up. 😱